### PR TITLE
Feature: Add support for animated avatars and profile backgrounds

### PR DESCRIPTION
**Pull request by sapsanDev**
Changes to data fetching in `fetchIdentifier()`:

- Fetching both backgrounds and animated avatar & frame from _IPlayerService/GetProfileItemsEquipped/v1/ (no authentication required)_

Changes to the `UserObject`:

- Changed `profile` to an object containing the the profile URL in `profile.url`
- Added `profile.background` and `profile.background_mini` corresponding to the user's profile/mini-profile background image
- Added animated avatar under `avatar.animated`
- Added avatar frame under `avatar.frame`
- Both backgrounds, avatar, and frame has `static` and `movie` keys, corresponding to the file type.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -59,26 +59,49 @@ class SteamAuth {
...
         );
+
         const players =
           response.data &&
           response.data.response &&
           response.data.response.players;
 
-        if (players && players.length > 0) {
+        if (players && players.length > 0)
+        {
```
Comment (LeeviHalme) [inline index.js:69]: Nit: The brace style here doesn't match the rest of the codebase. Use `if (condition) { ... }` to keep it consistent.

```suggestion
        if (players && players.length > 0) {
```

File: index.js [hunk 2]
```diff
@@ -59,26 +59,49 @@ class SteamAuth {
...
+            AnimatedAvatar.data.response.avatar;
+
+          const AvatarFrame = await axios.get(
+            `https://api.steampowered.com/IPlayerService/GetAvatarFrame/v1/?key=${this.apiKey}&steamid=${steamId}`
+          );
+
+          const Frame =
+            AvatarFrame.data &&
+            AvatarFrame.data.response &&
+            AvatarFrame.data.response.avatar_frame;
```
Comment (LeeviHalme) [inline index.js:89]: Nit: Variable naming uses `camelCase` style elsewhere.

```suggestion
          const animatedAvatar = await axios.get(
            `https://api.steampowered.com/IPlayerService/GetAnimatedAvatar/v1/?key=${this.apiKey}&steamid=${steamId}`
          );

          const animated = 
            animatedAvatar.data &&
            animatedAvatar.data.response &&
            animatedAvatar.data.response.avatar;

          const avatarFrame = await axios.get(
            `https://api.steampowered.com/IPlayerService/GetAvatarFrame/v1/?key=${this.apiKey}&steamid=${steamId}`
          );

          const frame =
            avatarFrame.data &&
            avatarFrame.data.response &&
            avatarFrame.data.response.avatar_frame;

```

File: index.js [hunk 3]
```diff
@@ -59,26 +59,49 @@ class SteamAuth {
...
           // Return user data
           resolve({
             _json: player,
             steamid: steamId,
             username: player.personaname,
             name: player.realname,
             profile: player.profileurl,
-            avatar: {
+            avatar:
+            {
```
Comment (LeeviHalme) [inline index.js:99]: Same with the brace style here.

```suggestion
            avatar: {
```

File: index.js [hunk 4]
```diff
@@ -59,26 +59,49 @@ class SteamAuth {
...
               small: player.avatar,
               medium: player.avatarmedium,
-              large: player.avatarfull
+              large: player.avatarfull,
+              animated: Animated.image_small ? `https://cdn.akamai.steamstatic.com/steamcommunity/public/images/${Animated.image_small}`: player.avatarfull,
+              frame: Frame.image_small ? `https://cdn.akamai.steamstatic.com/steamcommunity/public/images/${Frame.image_small}` : null
```
Comment (LeeviHalme) [inline index.js:104]: Add safe accessing to the object.

```suggestion
              large: player.avatarfull,
              animated: animated?.image_small ? `https://cdn.akamai.steamstatic.com/steamcommunity/public/images/${Animated.image_small}`: player.avatarfull,
              frame: frame?.image_small ? `https://cdn.akamai.steamstatic.com/steamcommunity/public/images/${Frame.image_small}` : null
```

**Review by LeeviHalme — CHANGES_REQUESTED** [review]
Thanks @sapsanDev for the new feature PR! Highly appreciated!

Requested changes for few nits that could be corrected to keep consistency, but other than that LGMT :)

File: index.js (hunk 4 above)
Comment (sapsanDev) [reply index.js]: Ok, ready.

**Review by LeeviHalme — APPROVED** [review]
Thanks! I've made a few commits myself for small style changes.

I've left out the `d.ts` file from this merge, but feel free to open another PR for the package providing its own types.

**Pull request merged**